### PR TITLE
Include JSON dataset in the demo-project

### PR DESCRIPTION
**Pull request by SajidAlamQB**
## Description
Related to: https://github.com/kedro-org/kedro-viz/issues/1929

## Development notes
Replaced `companies.csv` with `companies_json.json` and added a new node that converts the json into a csv. This should make the json pre-viewable in the demo.

This is what it looks like with the new `companies_json.json` in the view 

Edit: 

After reviewing with Rashida we now have a new reporting node `get_top_shuttles_data` that takes the head of the input table and converts it into JSON. See below for results:

image

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Comment by SajidAlamQB** [comment]
> I see that #1929 suggests to replace companies to json. I think it would be nice to introduce a new dataset rather than modifying existing datasets in the demo_project.

I'm not opposed but I feel like adding another dataset would be a bit random. If we wanted to include it for this demo we would have to rework it to make it feel natural.

**Review by tynandebold — APPROVED** [review]

**Review by jitu5 — APPROVED** [review]

File: demo-project/conf/base/catalog_01_raw.yml [hunk 1]
```diff
@@ -1,3 +1,7 @@
+companies_json:
+  type: json.JSONDataset
+  filepath: ${_base_location}/01_raw/companies_json.json
+
 companies:
```
Comment (ravi-kumar-pilla) [inline demo-project/conf/base/catalog_01_raw.yml:5]: If we plan on having only json, can we remove the csv entry in catalog ?

**Review by Huongg — APPROVED** [review]
LGTM, thanks @SajidAlamQB

**Comment by SajidAlamQB** [comment]
After reviewing with Rashida we now have a new reporting node `get_top_shuttles_data` that takes the head of the input table and converts it into JSON. See below for results:

image

**Review by rashidakanchwala — APPROVED** [review]

**Pull request merged**